DFC.ServiceTaxonomy.PageLocation/Drivers/PageLocationPartDisplayDriver.cs

```diff
@@ -112,6 +115,39 @@ private async Task ValidateAsync(PageLocationPart pageLocation, IUpdateModel upd
                     }
                 }
             }
+
+            if (!string.IsNullOrWhiteSpace(pageLocation.FullUrl))
+            {
+                ContentItem? taxonomy = await _session.Query<ContentItem, ContentItemIndex>(x =>
+                    //TODO: get rid of magic strings?
+                    x.ContentType == "Taxonomy" && x.DisplayText == "Page Locations" && x.Latest && x.Published).FirstOrDefaultAsync();
+
+                if (taxonomy != null)
+                {
+                    RecursivelyBuildUrls(taxonomy);
+
+                    if (PageLocations.Any(x => x.Equals(pageLocation.FullUrl.Trim('/'), StringComparison.OrdinalIgnoreCase)))
+                    {
+                        updater.ModelState.AddModelError(Prefix, nameof(pageLocation.FullUrl), "This URL has already been used as a Page Location");
+                    }
+                }
+            }
+        }
+
+        private List<string> PageLocations = new List<string>();
+
+        private void RecursivelyBuildUrls(ContentItem taxonomy)
```

Comment:
Think you _could_ use morelinq's TraverseDepthFirst or TraverseBreadthFirstHere instead.